src/modules/fundingManager/bondingCurveFundingManager/BondingCurveFundingManagerBase.sol

```diff
@@ -0,0 +1,143 @@
+    Module
+{
+    using Strings for uint;
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+    uint8 internal tokenDecimals;
+    bool internal buyIsOpen;
+    uint internal buyFee;
+    address internal collateral;
+    // Q: Do we want to set other beneficiary in the contract then the module itself for the fees?
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    function token() public view returns (IERC20) {
+        return __Module_orchestrator.token();
+    }
+
+    function buyOrder(uint _depositAmount) external payable virtual {
+        // WiP
+        // Deduct fee from incoming value. Fee is paid in collateral token
+        _issueTokens(_depositAmount, collateral);
+    }
+
+    function openBuy() external onlyOrchestratorOwnerOrManager {
+        // Function to set the PAMM buy functionality to open
+        _openBuy();
+    }
+
+    function closeBuy() external onlyOrchestratorOwnerOrManager {
+        // Function to set the PAMM buy functionality to close
+        _closeBuy();
+    }
+
+    function updateBuyFee(uint _fee) external onlyOrchestratorOwnerOrManager {
+        // Should update the buy fee of the contract
+        _updateBuyFee(_fee);
+    }
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    function decimals()
+        public
+        view
+        override(ERC20Upgradeable)
+        returns (uint8)
+    {
+        return tokenDecimals;
+    }
+
+    function deposit(uint amount) external {}
+
+    function depositFor(address to, uint amount) external {}
+
+    function withdraw(uint amount) external {}
+
+    function withdrawTo(address to, uint amount) external {}
+
+    //--------------------------------------------------------------------------
+    // Internal Functions Implemented in Downstream Contract
+    function _issueTokensFormulaWrapper(
+        uint _depositAmount,
+        address _collateral
+    ) internal virtual returns (uint);
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    function _openBuy() internal {
+        // Validate if not open
+        buyIsOpen = true;
+    }
+
+    function _closeBuy() internal {
+        // Validate if not closed
+        buyIsOpen = false;
+    }
+
+    function _setCollateral(address _collateral) internal {
+        collateral = _collateral;
+    }
+
+    function _updateBuyFee(uint _fee) internal {
+        buyFee = _fee;
+    }
+
+    function _issueTokens(uint _depositAmount, address _collateral)
+        internal
+        returns (uint mintAmount)
+    {
+        mintAmount = _issueTokensFormulaWrapper(_depositAmount, _collateral);
+        _mint(msg.sender, mintAmount);
+    }
+
+    function _setTokenDecimals(uint8 _decimals) internal {
+        // Input check
```

Comment:
What's the thought behind this? This would be the decimals of the token we mint and give out, right? Is there a reason we wouldn't want to stick with 18?

Reply:
I brought this up in a call with Felix. He suggested that I could stick to what was implemented before and follow the same pattern. To be more specific, the RebaseFundingManager also enables the setting of decimals so I decided to do the same

Reply:
Ok yeah, makes sense to keep it consinstent